### PR TITLE
tahoe-lafs: 1.13.0 -> unstable-2019-04-08

### PR DESCRIPTION
**Pull request by manveru**
###### Motivation for this change

The new version has a lot of improvements and is required to get Gridsync to work.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/networking/p2p/tahoe-lafs/default.nix [hunk 1]
```diff
@@ -1,65 +1,60 @@
...
+    testtools
+    fixtures
   ];
 
-  checkInputs = with pythonPackages; [ hypothesis twisted ];
+  # Tests are usually run via tox, but we don't need this overhead.
+  # Still a lot of tests are failing, need some help to debug this.
+  # checkPhase = ''
+  #   trial --rterrors allmydata
+  # '';
```
Comment (dotlambda) [inline pkgs/tools/networking/p2p/tahoe-lafs/default.nix]: Please keep the `checkPhase` and set `doCheck = false`.
We might be able to help if you post the output of a test run.

File: pkgs/tools/networking/p2p/tahoe-lafs/default.nix [hunk 2]
```diff
@@ -1,65 +1,60 @@
...
   namePrefix = "";
 
-  src = fetchurl {
-    url = "https://tahoe-lafs.org/downloads/tahoe-lafs-${version}.tar.bz2";
-    sha256 = "11pfz9yyy6qkkyi0kskxlbn2drfppx6yawqyv4kpkrkj4q7x5m42";
+  src = fetchFromGitHub {
+    owner = "tahoe-lafs";
+    repo = "tahoe-lafs";
+    rev = "325c522d7c73065e3f7e10518a9785b822d1405a";
+    sha256 = "0000000000000000000000000000000000000000000000000000";
```
Comment (dotlambda) [inline pkgs/tools/networking/p2p/tahoe-lafs/default.nix]: ??

File: pkgs/tools/networking/p2p/tahoe-lafs/default.nix [hunk 3]
```diff
@@ -1,65 +1,60 @@
...
-# plugins, because it tries to write them into Twisted's (immutable)
-# store path. The problem appears to be non-fatal, but there's probably
-# some loss of functionality because of it.
+{ fetchFromGitHub, lib, unzip, nettools, pythonPackages, texinfo, sqlite }:
 
 pythonPackages.buildPythonApplication rec {
-  version = "1.13.0";
+  version = "unstable-2019-04-08";
   name = "tahoe-lafs-${version}";
   namePrefix = "";
```
Comment (dotlambda) [inline pkgs/tools/networking/p2p/tahoe-lafs/default.nix]:
```suggestion
```

File: pkgs/development/python-modules/txaio/default.nix [hunk 4]
```diff
@@ -18,7 +18,7 @@ buildPythonPackage rec {
   '';
 
   # Needs some fixing for 3.7
-  doCheck = !isPy37;
+  doCheck = !isPy3k;
```
Comment (dotlambda) [inline pkgs/development/python-modules/txaio/default.nix]: Why did you change this?

File: pkgs/development/python-modules/subunitreporter/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  # postPatch = ''
+  #   substituteInPlace setup.py \
+  #     --replace "boltons >= 19.0.1" boltons
+
+  #   # depends on eliot.prettyprint._main which we don't have here.
+  #   rm eliot/tests/test_prettyprint.py
+  # '';
+
+  buildInputs = [ ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/subunitreporter/default.nix]:
```suggestion
```

File: pkgs/development/python-modules/subunitreporter/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    sha256 = "0vdmpp5hln38npn173bfjnqhd9dapxv9r30mi3sfdgsdmy7f4ynn";
+  };
+
+  # postPatch = ''
+  #   substituteInPlace setup.py \
+  #     --replace "boltons >= 19.0.1" boltons
+
+  #   # depends on eliot.prettyprint._main which we don't have here.
+  #   rm eliot/tests/test_prettyprint.py
+  # '';
```
Comment (dotlambda) [inline pkgs/development/python-modules/subunitreporter/default.nix]: Why is this commented out?

File: pkgs/development/python-modules/subunitreporter/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{ stdenv, buildPythonPackage, fetchPypi
+, zope_interface
+, twisted
+, python-subunit
+
+, pyrsistent
+, boltons
+, hypothesis
+, testtools
+, pytest
```
Comment (dotlambda) [inline pkgs/development/python-modules/subunitreporter/default.nix]: Most of these aren't used.

File: pkgs/development/python-modules/python-subunit/default.nix [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+buildPythonPackage rec {
+  pname = "python-subunit";
+  version = "1.3.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1fsw8rsn1s3nklx06mayrg5rn2zbky6wwjc5z07s7rf1wjzfs1wn";
+  };
+
+  checkPhase = "true"; # tests require internet
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-subunit/default.nix]:
```suggestion
  # tests require internet
  doCheck = false;
```

File: pkgs/development/python-modules/python-subunit/default.nix [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, buildPythonPackage, fetchPypi
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-subunit/default.nix]:
```suggestion
{ lib, buildPythonPackage, fetchPypi
```

File: pkgs/development/python-modules/eliot/default.nix [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "boltons >= 19.0.1" boltons
+
+    # depends on eliot.prettyprint._main which we don't have here.
+    rm eliot/tests/test_prettyprint.py
+  '';
+
+  buildInputs = [ zope_interface hypothesis ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/eliot/default.nix]: Why are these `buildInputs`?

File: pkgs/development/python-modules/eliot/default.nix [hunk 11]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, buildPythonPackage, fetchPypi
```
Comment (dotlambda) [inline pkgs/development/python-modules/eliot/default.nix]:
```suggestion
{ lib, buildPythonPackage, fetchPypi
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please make a separate commit for each package.

**Comment by dotlambda** [comment]
@manveru Did you ask upstream to make a release or do you know when to expect one?

File: pkgs/development/python-modules/txaio/default.nix (hunk 4 above)
Comment (manveru) [reply pkgs/development/python-modules/txaio/default.nix]: It didn't pass tests on python 3.8, and `isPy38` doesn't exist.

**Comment by manveru** [comment]
@dotlambda so, this was mostly for @MostAwesomeDude to take a look at, not a final PR. Still thanks for all the feedback, will try to address it when I get a bit more time.

I asked them about a new release when I reported excessive memory and CPU usage. They don't know yet when it'll happen, so I thought we can just as well get this updated for now.

Maybe we should make a separate package for it?

**Comment by dotlambda** [comment]
> this was mostly for @MostAwesomeDude to take a look at, not a final PR

This should be made obvious from the PR title/description.

> Maybe we should make a separate package for it?

We can decide this once the PR is ready.

**Comment by manveru** [comment]
@dotlambda not going to complain, I'm actually happy this got reviewed, usually my PRs sit around for a few weeks unless I ping someone specifically ;)

**Comment by exarkun** [comment]
Not sure what test failures you saw, but looking at results I get locally, some of the test failures are due to some test files not being installed.  This is fixable in Tahoe-LAFS itself with this patch:

```
diff --git a/setup.py b/setup.py
index 361e2c5dc..2e7cabfb4 100644
--- a/setup.py
+++ b/setup.py
@@ -249,7 +249,7 @@ setup(name="tahoe-lafs", # also set in __init__.py
                 "test": PleaseUseTox,
                 },
       package_dir = {'':'src'},
-      packages=find_packages('src'),
+      packages=find_packages('src') + ['allmydata.test.plugins'],
       classifiers=trove_classifiers,
       python_requires="<3.0",
       install_requires=install_requires,

```

**Comment by dotlambda** [comment]
> some of the test failures are due to some test files not being installed

we don't want to install those as they are just for testing

**Comment by FRidh** [comment]
Nothing happening so closing.
